File: TEO/Mav.UI/Mav.Ui/src/app/services/admin/i-settings.service.ts
import { Observable } from 'rxjs/Observable';
import { GenericResult } from '../../models/generic.result.model';
import { SettingViewModel } from '../../models/admin/setting.view.model';

export interface ISettingsService {

    getSettingById(id: number): Observable<SettingViewModel>;

    getSettingByName(name: string): Observable<SettingViewModel>;

    getSettings(): Observable<SettingViewModel[]>;

    updateSetting(setting: SettingViewModel): Observable<GenericResult<number>>;

}

File: TEO/Mav.Services/Abstract/ISettingsService.cs
using Mav.Common.Models;
using Mav.Models.ViewModels;
using System.Collections.Generic;
using System.Threading.Tasks;

namespace Mav.Services.Abstract
{
    public interface ISettingsService
    {
        Task<T> GetSettingById<T>(int id);

        Task<T> GetSettingByName<T>(string name);

        Task<List<T>> GetSettings<T>();

        Task<GenericResult<T>> UpdateSetting<T>(SettingViewModel m);
    }
}
